account_asset_management/migrations/12.0.1.0.0/post-migration.py

```diff
@@ -18,20 +18,44 @@ def adjust_asset_values(env):
         FROm account_asset_profile aap
         WHERE aa.profile_id = aap.id""",
     )
-    # Adjust method_time, method_number and method_period
-    number = sql.Identifier(openupgrade.get_legacy_name('method_number'))
-    period = sql.Identifier(openupgrade.get_legacy_name('method_period'))
+    # Adjust method_time, method_end, method_number and method_period
+    method_number = sql.Identifier(openupgrade.get_legacy_name('method_number'))
+    method_period = sql.Identifier(openupgrade.get_legacy_name('method_period'))
+    method_time = sql.Identifier(openupgrade.get_legacy_name('method_time'))
     for table in ['account_asset_profile', 'account_asset']:
         table = sql.Identifier(table)
         openupgrade.logged_query(
             env.cr, sql.SQL("""
             UPDATE {table}
             SET method_time = 'year',
+                method_end = NULL,
                 method_number = ({number} * {period}) / 12
-            WHERE MOD({number} * {period}, 12) = 0
+            WHERE MOD({number} * {period}, 12) = 0 AND {time} != 'end'
             """).format(
-                number=number,
-                period=period,
+                number=method_number,
+                period=method_period,
+                time=method_time,
+                table=table,
+            ),
+        )
+        openupgrade.logged_query(
```

Comment:
I still think you should remove this

Reply:
and don't restrict the previous one

Reply:
give reasons, please.

Reply:
- Don't be destructive
- Let the method as it was. Maybe one day an extension module cover such case.
- You are not providing a good alternative
- It doesn't hurt as it was, as the depreciation table is already computed.

Reply:
- I am not destructive, the columns are copied in pre-migration.
 - How should that be okey? A client have some records with the 'end' method. I migrate and happens this records are unusable because the 'end' method doesn't exist. Oh, should I make the module extension then? why? Be serious here. You introduced a query that made some records with 'number' method became 'year' method, with the comment in the code "Avoid method_time != 'year' when possible + adjust other values in assets and asset profiles." So changing 'number' records are okey but changing 'end' records are not okey? It's hilarious.
 - Which alternative should I provide? why? who invented this 'year' method what intended is to fuse the 'end' method (using dates) with the 'number' method (using number) in one method. Just so happens some 'number' records cannot be transformed to 'year' method, that why we reintroduce 'number' method. But the 'end' method transforms correctly to 'year' method, that why nobody has proposed to reintroduce 'end' method, and this is why the code doesn't support in anymore.

Reply:
OK, I abstain here then, as I don't have time to verify what you say, that it may be correct. My reluctant was for putting as `year` and 0 if there's a conversion to other one. As I don't have such case and my customers are already on v12, it's not a problem for me.